src/Sarif/Visitors/SortingVisitor.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.Sarif.Visitors
+{
+    public class SortingVisitor : SarifRewritingVisitor
+    {
+        private readonly IDictionary<int, int> ruleIndexMap;
```

Comment:
>IDictionary [](http://example.com/codeflow?start=25&length=11)

Stop using IDictionary if you want a perf micro-optimization (so you can use TryUpdate...)

Reply:
changed to Dictionary